learning_wavelets/evaluation_scripts/exact_recon_unet_evaluate.py

```diff
@@ -0,0 +1,50 @@
+from learning_wavelets.config import CHECKPOINTS_DIR
+from learning_wavelets.data.datasets import im_dataset_bsd68
+from learning_wavelets.evaluate import METRIC_FUNCS
+from learning_wavelets.evaluate import Metrics 
+from learning_wavelets.models.exact_recon_unet import ExactReconUnet
+import tensorflow as tf
+from tqdm import tqdm
+
+
+def evaluate_unet(
+        noise_std_test=30,
+        run_id='ExactReconUnet_4_bsd500_0_55_2000_1620137581',
+        n_epochs=500,
+        n_output_channels=1,
+        kernel_size=3,
+        layers_n_channels=[64, 128, 256, 512, 1024],
+        layers_n_non_lins=2,
+        non_linearity='relu',
+    ):
+    
+
+    run_params = {
+        'n_output_channels': n_output_channels,
+        'kernel_size': kernel_size,
+        'layers_n_channels': layers_n_channels,
+        'layers_n_non_lins': layers_n_non_lins,
+        'non_linearity': non_linearity,
+    }
+
+    val_set = im_dataset_bsd68(
+        mode='testing',
+        batch_size=1,
+        patch_size=None,
+        noise_std=noise_std_test,
+        return_noise_level=True,
+    )
+
+    
+    model = ExactReconUnet(**run_params)
+
+    inputs = [tf.zeros((1, 32, 32, 1)), tf.zeros((1, 1))]
+    model(inputs)
+    
+    model.load_weights(f'{CHECKPOINTS_DIR}checkpoints/{run_id}-{n_epochs}.hdf5')
+    
+    eval_res = Metrics(METRIC_FUNCS)
+    for x, y_true, size in tqdm(val_set.as_numpy_iterator()):
+        y_pred = model.predict(x)
+        eval_res.push(y_true[..., 0], y_pred[..., 0])
```

Comment:
I think here you don't need to slice the last dimension
```suggestion
        eval_res.push(y_true, y_pred)
```

But maybe check that it makes sense before applying the change